_docs/versioning.md

```diff
@@ -0,0 +1,87 @@
+
+1. **Clarity over Convention**: While we respect semantic versioning principles, we've adapted them to better communicate what matters most to our apps - distinguishing between main releases and hotfixes.
+
+2. **Go Module Compatibility Challenges**: Go's module system requires path changes for major version increments (v2+), which adds unnecessary complexity for our use case since we're primarily building a library for our own applications.
+
+3. **Release Coordination**: Our versioning system is designed to facilitate coordination between status-go and the apps that depend on it, prioritizing clear communication about changes over strict semantic versioning rules.
+
+4. **Simplicity**: By keeping the MAJOR version fixed and focusing on MINOR for main releases and PATCH for hotfixes, we maintain a simpler, more predictable versioning system for our teams.
+
+## Current Versioning Approach
+
+We use a simplified versioning approach based on [Semantic Versioning](https://semver.org/):
+
+- We keep the MAJOR version fixed (we no longer bump it for breaking changes)
+- We bump MINOR version for releases from the main branch
+- We bump PATCH version for hotfixes on release branches
+
+Version numbers are formatted as `vX.Y.Z` (e.g., `v10.26.0`).
+
+> 💡 Yes, we may end up with versions like `v10.250.0` eventually. This is by design and works well for our use case.
+
+## Tagging Versions
+
+1. We use `git` tags to track the version of the library
+2. Run `./_assets/scripts/version.sh` to get the current version
+3. To create a new version tag:
+   - Run `make tag-version` to create a tag for `HEAD`, or
+   - Run `make tag-version TARGET_COMMIT={hash}` to create a tag for a specific hash
+4. Don't forget to `git push origin {tag_created}` to publish the tag
+5. The created tag can be used to link mobile client to the new version
+
+## Release Branches
+
+- Release branches follow the format `release/vX.Y.x` (e.g., `release/v10.26.x`)
+- We create these branches as needed for desktop/mobile releases
+- Tags on `develop` branch (with PATCH=0) are made as needed - this is used by mobile team who reference status-go by version tag
+- Tags on `release/*` branches (with PATCH>0) are created manually when a patch release is needed for desktop/mobile
+
+## Context & History
+
+Originally, we used semantic versioning with MAJOR version 1, incrementing the MAJOR version for every breaking change ([PR #5829](https://github.com/status-im/status-go/pull/5829)). However, Go modules require that packages with a MAJOR version greater than 1 include a `/v{MAJOR}` suffix in their import paths. We decided against this approach because `status-go` is not intended to be used as an importable Go package, but rather as a shared library for [status-desktop](https://github.com/status-im/status-desktop) and [status-mobile](https://github.com/status-im/status-mobile). [(Issue #6049)](https://github.com/status-im/status-go/issues/6049)
```

Comment:
>"but rather as a shared library for status-desktop and status-mobile

That's not really true now, is it? Since Mobile is going into maintenance mode does it make sense to make these kind of changes if Desktop will now be the only consumer of `status-go`?

Reply:
@jakubgs This is not really a change, more of documenting a fact 😄 

1. For now Mobile is still going to get updates, and I assume will get `status-go` updates as well. So we'll still want to tag versions for it.
2. Even if it's only used in `status-desktop`, we still need to define _some_ versioning. Even if it means never tag a version again and just use commits.

I think what's described here should work for a few months. And we'll need to update this instructions when mobile repo is archived.

Reply:
Honestly, if just desktop would be consuming it then I'd just follow the same versioning as Desktop itself for simplcity.